pytest_container/container.py

```diff
@@ -871,11 +872,89 @@ def prepare_container(
             )
 
 
+@dataclass(frozen=True)
+class ContainerRemoteEndpoint:
+    _container_id: str
+    _runtime: OciRuntimeBase
+
+    def __post_init__(self) -> None:
+        assert self._container_id, "Container ID must not be empty"
+
+    def check_output(self, cmd: str, strip: bool = True) -> str:
+        """Run a command in the container and return its output."""
+        return self._runtime.run_command(
+            "exec",
+            self._container_id,
+            "/bin/sh",
+            "-c",
+            cmd,
+            strip=strip,
+        )
+
+    def exists(self, command: str) -> bool:
+        """Check if a command exists in the container."""
+        return (
+            self.check_output(f"command -v {command} || echo '<missing>'")
+            != "<missing>"
+        )
+
+    def file(self, path: str) -> "ContainerConnectionFile":
+        return ContainerConnectionFile(path=path, _remote=self)
+
+    def copy(
+        self, local: Union[str, Path], remote: str
+    ) -> "ContainerConnectionFile":
+        """Copy a file from the host to the container."""
+        if isinstance(local, Path):
+            local = str(local.absolute())
+        self._runtime.run_command(
+            "cp", local, f"{self._container_id}:{remote}"
+        )
+        return self.file(remote)
+
+
+@dataclass(frozen=True)
+class ContainerConnectionFile:
+    path: str
+    _remote: ContainerRemoteEndpoint
+
+    def _test(self, test: str) -> bool:
+        return (
+            self._remote.check_output(
+                f"test {test} {self.path} && echo 1 || echo 0"
+            )
+            == "1"
+        )
+
+    @property
+    def exists(self) -> bool:
+        return self._test("-e")
+
+    @property
+    def is_file(self) -> bool:
+        return self._test("-f")
+
+    @property
+    def is_directory(self) -> bool:
+        return self._test("-d")
+
+    @property
+    def content_string(self) -> str:
+        try:
+            return self._remote.check_output(f"cat {self.path}", strip=False)
+        except subprocess.CalledProcessError as e:
+            if "Is a directory" in e.stderr:
+                raise ValueError(f"{self.path} is a directory") from e
+            raise
+
+    def listdir(self) -> List[str]:
```

Comment:
I don't think that it is maintainable to start implementing a subset of testinfra in pytest_container. I think all that we should offer is `check_output`. If a user doesn't wish to use testinfra, then that's fine by me, but then they should handle these checks by themself.

Reply:
This is only the set that is needed in the test suite. I can change it to private API. I personally, don't have a use for any of it.